server/sonar-server/src/main/java/org/sonar/server/issue/notification/NewIssuesEmailTemplate.java

```diff
@@ -58,44 +71,111 @@ public EmailMessage format(Notification notification) {
     }
     String projectName = notification.getFieldValue(FIELD_PROJECT_NAME);
 
-    StringBuilder sb = new StringBuilder();
-    sb.append("Project: ").append(projectName).append("\n\n");
-    sb.append(notification.getFieldValue("count")).append(" new issues").append("\n\n");
-    sb.append("   ");
+    StringBuilder message = new StringBuilder();
+    message.append("Project: ").append(projectName).append("\n\n");
+    appendSeverity(message, notification);
+    appendAssignees(message, notification);
+    appendTags(message, notification);
+    appendComponents(message, notification);
+    appendFooter(message, notification);
+
+    return new EmailMessage()
+      .setMessageId("new-issues/" + notification.getFieldValue(FIELD_PROJECT_KEY))
+      .setSubject(projectName + ": " + notification.getFieldValue(METRIC.SEVERITY + ".count") + " new issues")
+      .setMessage(message.toString());
+  }
+
+  private void appendComponents(StringBuilder message, Notification notification) {
+    if (notification.getFieldValue(METRIC.COMPONENT + ".1.label") != null) {
+      message.append("   Components:\n");
+    } else {
+      return;
+    }
+
+    for (int i = 1; i <= 5; i++) {
+      String component = notification.getFieldValue(METRIC.COMPONENT + "." + i + ".label");
+      if (component != null) {
+        message.append("      ")
+          .append(component)
+          .append(" : ")
+          .append(notification.getFieldValue(METRIC.COMPONENT + "." + i + ".count"))
+          .append("\n");
+      } else {
+        break;
+      }
+    }
+  }
+
+  private void appendAssignees(StringBuilder message, Notification notification) {
+    if (notification.getFieldValue(METRIC.LOGIN + ".1.label") != null) {
+      message.append("   Assignee - ");
+    } else {
+      return;
+    }
+    for (int i = 1; i <= 5; i++) {
+      String login = notification.getFieldValue(METRIC.LOGIN + "." + i + ".label");
+      if (login != null) {
+        String name = userIndex.getNullableByLogin(login).name();
+        message.append(name)
+          .append(": ")
+          .append(notification.getFieldValue(METRIC.LOGIN + "." + i + ".count"));
+        if (i < 5) {
+          message.append("   ");
+        }
+      } else {
+        break;
+      }
+    }
+    message.append("\n");
+  }
+
+  private void appendTags(StringBuilder message, Notification notification) {
+    if (notification.getFieldValue(METRIC.TAGS + ".1.label") != null) {
+      message.append("   Tags - ");
+    } else {
+      return;
```

Comment:
yep, I wasn't satisfied either. At one point I said I must ship. I refactor this part